lms/djangoapps/commerce/signals.py

```diff
@@ -57,6 +58,30 @@ def handle_refund_order(sender, course_enrollment=None, **kwargs):
             )
 
 
+# pylint: disable=unused-argument
+@receiver(REVOKE_ENTITLEMENT)
+def handle_refund_entitlement(sender, course_entitlement=None, **kwargs):
+    if not is_commerce_service_configured():
+        return
+
+    if course_entitlement and course_entitlement.is_entitlement_refundable():
+        try:
+            request_user = get_request_user()
+            if request_user and course_entitlement.user == request_user:
+                refund_entitlement(course_entitlement)
+        except Exception as exc:  # pylint: disable=bare-except
+            # don't assume the signal was fired with `send_robust`.
+            # avoid blowing up other signal handlers by gracefully
+            # trapping the Exception and logging an error.
+            log.exception(
+                "Unexpected exception while attempting to initiate refund for user [%s], "
+                "course entitlement [%s] message: [%s]",
+                course_entitlement.user.id,
+                course_entitlement.uuid,
+                str(exc)
```

Comment:
Nice improvement on the handle_refund_order version of this (to record the exception string itself).